charon/src/bin/charon-driver/translate/translate_bodies.rs

```diff
@@ -566,28 +569,118 @@ impl BodyTransCtx<'_, '_, '_> {
                                 let tref = self.translate_trait_impl_expr(span, impl_expr)?;
-                                match &impl_expr.r#impl {
+                                let vtable = match &impl_expr.r#impl {
                                     hax::ImplExprAtom::Concrete(tref) => {
                                         // Ensure the vtable type is translated.
-                                        let _: GlobalDeclId = self.register_item(
+                                        Some(self.translate_item(
                                             span,
                                             tref,
                                             TransItemSourceKind::VTableInstance(
                                                 TraitImplSource::Normal,
                                             ),
-                                        );
+                                        )?)
                                     }
                                     // TODO(dyn): more ways of registering vtable instance?
                                     _ => {
                                         trace!(
                                             "impl_expr not triggering registering vtable: {:?}",
                                             impl_expr
-                                        )
+                                        );
+                                        None
+                                    }
+                                };
+                                UnsizingMetadata::VTableDirect(tref, vtable)
+                            }
+                            hax::UnsizingMetadata::NestedVTable(impl_expr) => {
+                                let (hax::TyKind::Ref(_, src_hax_ty, _)
+                                | hax::TyKind::RawPtr(src_hax_ty, _)) = src_hax_ty.kind()
+                                else {
+                                    raise_error!(
+                                        self,
+                                        span,
+                                        "Nested vtable unsizing only supported for ref types, got {:?}",
+                                        src_hax_ty
+                                    );
+                                };
+                                let hax::TyKind::Dynamic(self_ty, preds, region) =
+                                    src_hax_ty.kind()
+                                else {
+                                    raise_error!(
+                                        self,
+                                        span,
+                                        "Nested vtable unsizing only supported for dyn traits, got {:?}",
+                                        src_hax_ty
+                                    );
+                                };
+
+                                // Add a binder that contains the existentially quantified type.
```

Comment:
Can you use `translate_dyn_binder` here instead of managing the binder levels yourself?